image/tarexport/save.go

```diff
@@ -576,12 +586,10 @@ func (s *saveSession) saveConfigAndLayer(ctx context.Context, id layer.ChainID,
 		layerPath = filepath.Join(outDir, lDgst.Algorithm().String(), lDgst.Encoded())
 	}
 
-	if createdTime != nil {
-		for _, fname := range []string{outDir, layerPath} {
-			// todo: maybe save layer created timestamp?
-			if err := system.Chtimes(fname, *createdTime, *createdTime); err != nil {
-				return distribution.Descriptor{}, errors.Wrap(err, "could not set layer timestamp")
-			}
+	for _, fname := range []string{outDir, layerPath} {
+		// todo: maybe save layer created timestamp?
```

Comment:
This TODO was added back in 2015 by @tonistiigi in https://github.com/moby/moby/commit/01ba0a935b9227583ec05a96e85273cc2276bb93#diff-9ac2cf7826d0cfc8316ca97dceae47447f431bc510fd4a4622ebeb53cc74d08aR295

I wonder if it's still clear what it's about, or if we should just drop it 🤔

Reply:
Not quite sure. This just does the `nil` check once but shouldn't change the behavior here. This PR still defers on a timestamp methodology, other than fixing the issue with the particular assets identified in https://github.com/docker/cli/issues/5515.

The other approach here would be to do a full walk and set them all to the same value after assembling the `save` file.

Reply:
Yeah, was mostly wondering if we had a specific reason to preserve the layer timestamp, because if not, we might as well remove the TODO comment here.

Resetting to the image's timestamp (what I saw you did in this PR) makes sense to me; at least it makes things stable for the same image